### DISCUSSION TITLE
YoloV5 Hub results: why Pascal_VOC format?

### DISCUSSION THREAD

**Discussion by Niels-Vdk** [ultralytics/yolov5#9110] [Q&A] [answered]
Hello there!

Using YoloV5 hub:
I have been working on making an automatic script that saves images with the corresponding labels (.txt file in YOLO format) when the model doesn't detect enough objects (or when there's a lot of detections, I will also save the output).

I have noticed that for training, YoloV5 uses YOLO format (XYWH), but the detections are Pascal_VOC (XYXY)
**Is there a particular reason why the output is Pascal_VOC?**
Maybe I missed somewhere the option to automatically use YOLO format for outputs?

For anyone interested, the code I am using is:

```
image_w = cap.get(3)
image_h = cap.get(4)

def pascal_voc_to_yolo(x1, y1, x2, y2, image_w, image_h):
    return [((x2 + x1)/(2*image_w)), ((y2 + y1)/(2*image_h)), (x2 - x1)/image_w, (y2 - y1)/image_h]
```

For the detection:
```
df = results.pandas().xyxy[0]
df_dict = results.pandas().xyxy[0].to_dict(orient="records")

timestamp = (datetime.now().strftime('%Y-%m-%d_%H-%M-%S-%m'))


#Hook is always visible in the screen, so it should always be detected. If not: save image for training purposes
if ('Hook' in df['name'].values) == False:
        print("no hook detected")
        cv2.imwrite(NoHookFolder+timestamp, frame)
        
        YoloResults = []
        for value in df_dict:
            print(value)
            x1 = int(value['xmin'])
            y1 = int(value['ymin'])
            x2 = int(value['xmax'])
            y2 = int(value['ymax'])
            cs = value['class']
            
            X, Y, W, H = (pascal_voc_to_yolo(x1, y1, x2, y2, image_w, image_h))
            YoloResults.append(f"{cs} {X} {Y} {W} {H}") #important to make a list, as there can be other detections besides the hook, saving on labeling time.
        with open(NoHookFolderLabels+timestamp+".txt", "w") as f:
            for YoloResult in YoloResults:
                f.write(YoloResult)
                f.write('\n')
```

Thanks a bunch!

**Comment by glenn-jocher** [comment, marked as answer]
@Niels-Vdk PyTorch Hub detection outputs are available in a variety of formats in Detections() class L666-L669.

https://github.com/ultralytics/yolov5/blob/51c9f9229731021f55a9ceb9f9504abfc979a54b/models/common.py#L666-L669

**Comment by Niels-Vdk** [comment]
@glenn-jocher Thanks a lot! I didn't know about this. I don't yet understand how to activate those lines but I'll try and figure it out.

**Reply by glenn-jocher** [reply]
results.xywh etc.

**Reply by Niels-Vdk** [reply]
Thank you. Case closed :)
